### ISSUE TITLE
Can we have lr_scheduler as a key input for the trainer() function?

### ISSUE THREAD

**Issue by HelloWorldLTY** [Lightning-AI/lightning#18974]
### Description & Motivation

I think adding this feature is important for pytorch-lightning extension.

cc @borda

**Comment by awaelchli** [comment]
@HelloWorldLTY Did you know you can return the scheduler in `LightningModule.configure_optimizers()`?
https://lightning.ai/docs/pytorch/stable/common/lightning_module.html#configure-optimizers

**Comment by HelloWorldLTY** [comment]
Hi, thanks a lot. I implemented this function based on your reference.
